src/shared/lib/helpers/color.ts

```diff
@@ -62,29 +62,23 @@ export const rgb5BitToGBCHex = (
     .padStart(6, "0");
 };
 
-export const indexSpriteColour = (g: number, objPalette: ObjPalette) => {
-  if (g < 65) {
-    return 3;
-  }
-  if (g < 130) {
-    return objPalette === "OBP1" ? 2 : 3;
-  }
-  if (g < 205) {
-    return objPalette === "OBP1" ? 2 : 1;
-  }
-  return 0;
+//NOTE: this assumes GB Studio's sprite image requirements are met... https://www.gbstudio.dev/docs/assets/sprites#image-requirements
+export const indexSpriteColour = (g: number, objPalette: [number, number, number, number]) => {
+  if (g < 130) return objPalette[3];
+  if (g < 205) return objPalette[2];
+  return objPalette[1];
```

Comment:
This also fixes a bug. Because "white" is always used as transparency, only 3 colors can actually be mapped. The previous version of this function only returned 3 different numbers (either 0,1,3 or 0,2,3), but it did so with 4 different inputs. This resulted in editor previews of sprites set to OBP1 that were different than what you would see in builds. Of course, you could only see this preview in the sprite sheet select in the game world view, so I don't suppose most folks would have noticed.